test/rules/no-restricted-globals/misc.ts.lint

```diff
@@ -0,0 +1,21 @@
+function typeTest() {
```

Comment:
A few missing test cases:

* `let { foo = event } = bar` - should be a lint complaint
* `let { foo = (() => event)() } = bar;` - should be a lint complaint
* Function with a parameter of a banned global, and references to that parameter
* `instanceof` and `typeof` usages

Reply:
You are really experienced!